cli/Makefile

```diff
@@ -10,21 +10,33 @@ NAME=cft
 BUCKET=gs://cft-cli
 INT_TEST_DIR=./bpmetadata/int-test
 SCHEMA_DIR=./bpmetadata/schema
-PROTO_DIR=./bpmetadata/proto
+SRC_PROTO_DIR=./bpmetadata/proto
+PROTOC_DIR=./bpmetadata
 
 # Setup the -ldflags option for go build here, interpolate the variable values
 LDFLAGS=-ldflags "-X $(GITHUB_REPO)/cli/cmd.Version=$(VERSION)"
 
-# need an extra build command to embed an most up to date schema in the binary
 .PHONY: build
-build:
-	go run ./${SCHEMA_DIR} -output=${SCHEMA_DIR}
+build: build-schema
 	go build ${LDFLAGS} -o ${BUILD_DIR}/${NAME}
 
+.PHONY: build-schema
+build-schema:
+	go run ./${SCHEMA_DIR} -output=${SCHEMA_DIR}
+
 .PHONY: protoc
-protoc:
-	protoc --go_out=${PROTO_DIR}/out --go_opt=paths=source_relative ${PROTO_DIR}/*.proto
-	go build ${PROTO_DIR}/out/...
+protoc: protoc-deps protoc-gen
+
+.PHONY: protoc-gen
+protoc-gen:
+	protoc -I=${SRC_PROTO_DIR} --go_opt=paths=source_relative --go_out=${PROTOC_DIR} ${SRC_PROTO_DIR}/*.proto
+	protoc-go-inject-tag -input="${PROTOC_DIR}/*.pb.go"
+
+.PHONY: protoc-deps
+protoc-deps:
+	go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.28
+	go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.2
+	go install github.com/favadi/protoc-go-inject-tag@v1.4.0
```

Comment:
nit: It would be better to just run protoc within a docker container. Maybe we can add it to our tools image?

Reply:
I'm gonna do this in a follow-up PR

Reply:
Done with #1713